### PR TITLE
Add scripts to rule them all

### PR DESCRIPTION
**Pull request by jacksonj04**
These scripts simplify the job of setting up, updating and starting this service by doing things like downloading dependencies, running data migrations and starting the server in a consistent way.

As per [RFC 023](https://github.com/dxw/tech-team-rfcs/blob/master/rfc-023-use-scripts-to-rule-them-all.md)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,9 @@
 # Ignore bundler config.
 /.bundle
 
+# Ignore vendor
+/vendor/gems
```
Comment (erbridge) [inline .gitignore]: I think: this is a nonstandard place for gems and usually you use `vendor` to explicitly _include_ dependencies (such as ones that aren't package managed).

Comment (jacksonj04) [reply .gitignore]: You're absolutely right, and inspired me to investigate what was going on and fix my local bundler config.

File: script/test [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  # already been set to `test` almost always means this is being called on its
+  # own from a `development` environment.
+  export RAILS_ENV="test" RACK_ENV="test"
+
+  script/update
+fi
+
+echo "==> Running tests…"
+
+if [ -n "$1" ]; then
```
Comment (jcoglan) [inline script/test]: This is a common mistake I've seen in many of our scripts for running tests. `rspec`'s arguments are complex and sometimes require quoting, for example `rspec -e 'name of test'`. The correct thing to do here is forward all arguments to `rspec` by calling:

```
rspec "$@"
```

This makes this whole conditional block unnecessary.

**Comment by CristinaRO** [comment]
Running `script/bootstrap` yields a deprecation warning:
```
[DEPRECATED] The `--without` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set without 'production'`, and stop using this flag
```
Running `script/setup` creates then drops the databases, then exits with an error:
```
Caused by:
PG::ConnectionBad: FATAL:  database "team_dashboard_development" does not exist
/Users/cristinarotaru/projects/team-dashboard/db/schema.rb:16:in `block in <main>'
/Users/cristinarotaru/projects/team-dashboard/db/schema.rb:13:in `<main>'
-e:1:in `<main>'
Tasks: TOP => db:schema:load
```

Should the operations be reversed?

**Comment by jcoglan** [comment]
That change to `test` looks good to me :)

**Comment by CristinaRO** [comment]
And one more: running the specs fails because of a version mismatch, would it be ok to prefix with `bundle exec`?

**Pull request merged**